lib/sensors/temperature.cpp

```diff
@@ -0,0 +1,53 @@
+#include "temperature.hpp"
+
+namespace hyped::sensors {
+
+Temperature::Temperature(hyped::io::I2c &i2c, hyped::core::ILogger &log) : log_(log), i2c_(i2c)
+{
+}
+
+std::optional<std::uint16_t> Temperature::read()
+{
+  const auto Status_check = i2c_.readByte(kMux, kStatus);
+
+  if (Status_check) {
+    if (Status_check.value() == 0) {
+      const std::optional<std::uint8_t> tempurature_high = i2c_.readByte(kMux, kTempuratureHigh);
+      const std::optional<std::uint8_t> tempurature_low  = i2c_.readByte(kMux, kTempuratureLow);
+      // getting the high and low values from the register
+
+      if (tempurature_high.has_value() && tempurature_low.has_value()) {
+        std::uint16_t tempurature = (tempurature_high.value() << 8) | tempurature_low.value();
+        tempurature               = tempurature * 0.01;
+
+        return tempurature;
+      } else {
+        log_.log(hyped::core::LogLevel::kFatal, "Temperature read not implemented");
+        return std::nullopt;
+      }
+      // Reading from the Status register to see if the tempurature is ready to be read
+      // Need to have it do the correct effect and output a log of what occurred
+
+    } else {
+      log_.log(hyped::core::LogLevel::kFatal, "Temperature read not implemented");
+      return std::nullopt;
+    }
+  } else {
+    log_.log(hyped::core::LogLevel::kFatal, "Temperature read not implemented");
+    return std::nullopt;
+  }
+}
+
+core::Result Temperature::configure()
+{
+  const core::Result write_result = i2c_.writeByteToRegister(kMux, kControl, 1);
```

Comment:
I don't think we want to write a 1 to the CTRL reg since this would activate one_shot_ mode requiring us to re-configure before reading again. We want to write a `0x3C` (0b111100) since this activates continuous mode, allows to to carry out mulitiple reads in succession (from IF_ADD_INC) and sets the output data rate to 200hz (the highest possible). I previously tested with this setting and it gave us expected results.

Reply:
also don't directly write `1` or `0x3C`,  make a `static constexpr` and call this `kConfigurationSetting` with a little explanation as to why that number (feel free to use how I justified it above)